.github/workflows/markdown-lint.yml

```diff
@@ -0,0 +1,73 @@
+name: Markdown Linting
+
+on:
+  workflow_call:
+    secrets:
+      CHECKOUT_TOKEN:
+        required: false
+        description: >
+          The GitHub token to authenticate checkout. Pass in a GitHub personal access token if authenticated submodules
+          are used.
+
+    # When updating defaults here also update them in the `markdown-lint` action.
+    inputs:
+      cancel-workflow-on-failure:
+        description: When set to "true", it will cancel the current workflow run with all jobs if this workflow fails.
+        type: string
+        default: 'true'
+      config:
+        description: Path to a file to use for the base configuration object (defaults to "lombiq.markdownlint.json").
+        type: string
+        default: '.github/actions/markdown-lint/lombiq.markdownlint.json'
+      fix:
+        description: Whether to fix issues automatically for the rules that support it (any truthy value enables).
+        type: string
+        default: ''
+      globs:
+        description: Glob expression(s) of files to lint (newline-delimited).
+        type: string
+        default: '**/*.{md,markdown}'
+      separator:
+        description: String to use as a separator for the "globs" input (defaults to newline).
+        type: string
+        default: "\n"
+      timeout-minutes:
+        type: number
+        default: 1
+        description: Configuration for the timeout-minutes parameter of the workflow.
+
+jobs:
+  markdown-linting:
+    name: Markdown Linting
+    runs-on: ubuntu-22.04
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+    steps:
+      - name: Checkout
+        uses: Lombiq/GitHub-Actions/.github/actions/checkout@dev
+        with:
+          token: ${{ secrets.CHECKOUT_TOKEN }}
+
+      # This is a workaround to be able to check submodules too in the repository.
```

Comment:
Could you elaborate what's the problem that prevents from checking submodules?

Reply:
TBH I don't know, I started this workflow as a copy of `spelling`, but `yaml-lint` doesn't have this step. I'll remove it for now and do a test run in OSOCE to see if it works without it.

Reply:
Yeah, I see it was added 2 years ago when .github/workflows/spelling.yml was originally created. If the problem is that the submodules aren't initialized by default (so just empty directories) then the correct way to do this is a simple one-liner: `git submodule update --init`.